src/server/debug.rs

```diff
@@ -360,6 +371,97 @@ impl Debugger {
         Ok(())
     }
 
+    pub fn recover_all(&self, threads: usize, read_only: bool) -> Result<()> {
+        let db = &self.engines.kv;
+
+        println!("Calculating split keys...");
+        let split_keys = divide_db(db, threads).unwrap().into_iter().map(|k| {
+            let k = Key::from_encoded(keys::origin_key(&k).to_vec())
+                .truncate_ts()
+                .unwrap();
+            k.as_encoded().clone()
+        });
+
+        let mut range_borders = vec![b"".to_vec()];
+        range_borders.extend(split_keys);
+        range_borders.push(b"".to_vec());
+
+        let mut handles = Vec::new();
+
+        for thread_index in 0..range_borders.len() - 1 {
+            let db = Arc::clone(db);
+            let start_key = range_borders[thread_index].clone();
+            let end_key = range_borders[thread_index + 1].clone();
+            if start_key == end_key {
+                continue;
+            }
+
+            let thread = ThreadBuilder::new()
+                .name(format!("mvcc-recover-thread-{}", thread_index))
+                .spawn(move || {
+                    println!(
+                        "thread {}: started on range [\"{}\", \"{}\")",
+                        thread_index,
+                        escape(&start_key),
+                        escape(&end_key)
+                    );
+
+                    let mut mvcc_checker =
```

Comment:
this part is verbose with `recover_region`, you can extract them into `recover_range`